src/main/java/com/carrotcreative/recyclercore/widget/ProgressRecyclerViewLayout.java

```diff
@@ -117,4 +124,131 @@ public void stopScroll()
     {
         mRecyclerView.stopScroll();
     }
+
+    // ========================================== //
+    // =============  Error State =============== //
+    // ========================================== //
+
+    public boolean isShowingErrorState()
+    {
+        if(mErrorStateView != null && mErrorStateView.getVisibility() == View.VISIBLE)
+        {
+            return true;
+        }
+        return false;
+    }
+
+    public void setErrorStateEnabled(boolean show)
+    {
+        if(mErrorStateView == null)
+        {
+            throw new IllegalStateException("Trying to setErrorStateEnabled without setting the error state View.");
+        }
+
+        if(show)
+        {
+            if(! isShowingErrorState())
+            {
+                saveCurrentViewState();
+                mErrorStateView.setVisibility(VISIBLE);
+                mRecyclerView.setVisibility(GONE);
+                mProgressBar.setVisibility(GONE);
+            }
+        }
+        else
+        {
+            restorePreviousViewState();
+        }
+    }
+
+    public void setErrorStateEnabled(View errorView, boolean show)
+    {
+        if(mErrorStateView == null)
+        {
+            mErrorStateView = errorView;
+            mContainer.addView(mErrorStateView);
```

Comment:
Ah, I really like the decision you made here to make it simply get added to the root FrameLayout.  That makes this really flexible.

I was thinking we were going to have a FrameLayout container inside of our old Relative Layout centered...

This is much better :100: